### PR TITLE
Refactor User Form for creating and editing user with validation

### PR DESCRIPTION
**Pull request by chiahuang**
+Refactor UserForm for user creation and modification

### CODE REVIEW

File: userapp/src/components/user/UserForm.js [hunk 1]
```diff
@@ -35,49 +60,43 @@ const UserForm = ({ initialUser, onSubmit}) => {
...
-    //                 setErrors({ ...user, [name]: ""})
-    //             } else {
-    //                 setErrors({ ...user, [name]: "Phone must have length of 10"})
-    //             }
-    //             break
-    //         default:
-    //             break
-    //     }
-    // }
+    const validate = (name, value) => {
```
Comment (david-geisinger) [inline userapp/src/components/user/UserForm.js:70]: we can simplify these cases to all call one method of validation (code simplification)

File: userapp/src/components/user/UserForm.js [hunk 2]
```diff
@@ -90,65 +109,76 @@ const UserForm = ({ initialUser, onSubmit}) => {
...
             <Grid container
-                direction="column"
-                justify="center"
-                alignItems="center">
+                direction="row"
+                justify="flex-start"
+                alignItems="flex-start"
+                spacing={1}>
                 <Grid item xs={6}>
                     <HelixTextField
```
Comment (david-geisinger) [inline userapp/src/components/user/UserForm.js:120]: first name and last name should be simple to combine into one call as well

File: userapp/src/components/user/UserForm.js [hunk 3]
```diff
@@ -90,65 +109,76 @@ const UserForm = ({ initialUser, onSubmit}) => {
...
+                    name='DateOfBirth'
+                    label='Date Of Birth'
+                    type="date"
+                    value={user.DateOfBirth}
+                    InputLabelProps={{ shrink: true }}
+                    onChange={handleInputChange}
                     />
-                    {/* {errors.DateOfBirth.length > 0 && <span className='error'>{errors.DateOfBirth}</span>} */}
                 </Grid>
+                {error.DateOfBirth.length > 0 && <span className='error'>{error.DateOfBirth}</span>}
```
Comment (david-geisinger) [inline userapp/src/components/user/UserForm.js:153]: why does this have special error case?

File: userapp/src/components/user/UserForm.js [hunk 4]
```diff
@@ -90,65 +109,76 @@ const UserForm = ({ initialUser, onSubmit}) => {
...
-                        variant="contained"
-                        type="cancel"
-                        size="large"
-                        href="/">
-                            Cancel
-                        </Button>
-                    </div>
+                <Grid item xs></Grid>
+                <Grid item xs={6} className={userFormClasses.buttonStyle}>
+                    <Button
```
Comment (david-geisinger) [inline userapp/src/components/user/UserForm.js:167]: do we have Helix submit button/helix save button/helix cancel button

File: userapp/src/components/user/UserForm.js (hunk 3 above)
Comment (chiahuang) [reply userapp/src/components/user/UserForm.js:153]: Meant to be deleted.

File: userapp/src/components/user/UserForm.js (hunk 2 above)
Comment (chiahuang) [reply userapp/src/components/user/UserForm.js:120]: Then, I will change the object on the bottom into Full Name instead

File: userapp/src/components/user/UserForm.js (hunk 4 above)
Comment (chiahuang) [reply userapp/src/components/user/UserForm.js:167]: Not yet

File: userapp/src/components/user/UserForm.js [hunk 5]
```diff
@@ -117,68 +154,34 @@ const UserForm = ({ initialUser, header, onSubmit}) => {
...
-                    error={error.FirstName.length !== 0}
-                    name='FirstName'
-                    label='First Name'
-                    value={user.FirstName}
-                    required={true}
-                    placeholder="John"
-                    helperText={error.FirstName}
-                    onChange={handleInputChange}
-                    />
+                    {setHelixTextField(error.FirstName.length !== 0, "FirstName", "First Name", user.FirstName, true, "Joe", error.FirstName, handleInputChange, {}, { maxLength: 26 })}
```
Comment (david-geisinger) [inline userapp/src/components/user/UserForm.js:157]: you should be able to do this passing in much less params (say 3-4)

**Review by david-geisinger — APPROVED** [review]

**Pull request merged**